crates/ruma-events/src/space/child.rs

```diff
@@ -0,0 +1,65 @@
+//! Types for the *m.space.child* event.
+
+use ruma_events_macros::EventContent;
+use ruma_identifiers::ServerNameBox;
+use serde::{Deserialize, Serialize};
+
+use crate::StateEvent;
+
+/// The admins of a space can advertise rooms and subspaces for their space by setting
+/// `m.space.child` state events. The `state_key` is the ID of a child room or space, and the
+/// content must contain a `via` key which gives a list of candidate servers that can be used to
+/// join the room
+pub type ChildEvent = StateEvent<ChildEventContent>;
+
+/// The payload for `ChildEvent`.
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+#[ruma_event(type = "m.space.child", kind = State)]
+pub struct ChildEventContent {
+    /// List of candidate servers that can be used to join the room. Children where `via` is not
+    /// present or invalid (not an array) are ignored.
```

Comment:
```suggestion
    /// List of candidate servers that can be used to join the room.
```

The "Children where `via` is not set" part is not relevant when you have an instance of this type, since that already presupposes deserialization having succeeded.

Reply:
Also, is there any reason this should be `Option<Vec<_>>` instead of `Vec<_>` is `None` is considered invalid anyways?

Reply:
The spec states

```
// no longer a child room
{
    "type": "m.space.child",
    "state_key": "!jklm:example.com",
    "content": {}
}
```

So I guess that removes the child from the space. That was the intention with the second sentence of the description

Reply:
Oh I think I get it. This is about redactions. Well redacted events are separate types in Ruma, so there is no need to make the redacted state representable in this type.

Reply:
I think it's simply about removing the relation from space to child. I guess someone would need to ask the spec people

Reply:
Yeah, it's about that but I'm pretty sure the means to achieve that is redacting the `m.space.child` event (not sending a new empty one).

Reply:
The spec does show an empty event tho? And it doesn't talk about redaction. @MTRNord Maybe you want to double check for us? :D

Reply:
Well a redacted event *is* an empty event (plus `redacted_because`). But it's been cleared up in the Spaces feedback room. It doesn't matter whether the latest `m.space.child` event is redacted or a new one with empty content is posted, both remove the child room from the space.

Reply:
… which means my recommendation of making the fields non-optional is not valid.

Reply:
Thanks for clearing that up